sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -368,79 +403,42 @@ private ResponseBodyMode determineResponseBodyMode(HttpRequest httpRequest, Http
-                        return doInputOutput(httpRequest, socket, sslSocketFactory);
-                    }
-
-                default:
-                    throw LOGGER.logThrowableAsWarning(
-                        new ProtocolException("Only HTTP and HTTPS are supported by this client."));
-            }
-        }
-
         /**
          * Calls buildAndSend to send a String representation of the request across the output stream, then calls
          * buildResponse to get an instance of HttpUrlConnectionResponse from the input stream
          *
          * @param httpRequest The HTTP Request being sent
-         * @param socket An instance of the SocketClient
+         * @param bufferedInputStream the input stream from the socket
+         * @param outputStream the output stream from the socket for writing the request
          * @return an instance of Response
          */
-        @SuppressWarnings("deprecation")
-        private static Response<?> doInputOutput(HttpRequest httpRequest, Socket socket,
-            SSLSocketFactory sslSocketFactory) throws IOException {
+        private static Response<?> sendPatchRequest(HttpRequest httpRequest, BufferedInputStream bufferedInputStream,
+            OutputStream outputStream) throws IOException {
             httpRequest.getHeaders().set(HttpHeaderName.HOST, httpRequest.getUrl().getHost());
-            if (!"keep-alive".equalsIgnoreCase(httpRequest.getHeaders().getValue(HttpHeaderName.CONNECTION))) {
-                httpRequest.getHeaders().set(HttpHeaderName.CONNECTION, "close");
-            }
+            OutputStreamWriter out = new OutputStreamWriter(outputStream);
 
-            try (BufferedReader in = new BufferedReader(
-                new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8));
-                OutputStreamWriter out = new OutputStreamWriter(socket.getOutputStream())) {
+            buildAndSend(httpRequest, out);
+            Response<?> response = buildResponse(httpRequest, bufferedInputStream);
+            HttpHeader locationHeader = response.getHeaders().get(HttpHeaderName.LOCATION);
+            String redirectLocation = (locationHeader == null) ? null : locationHeader.getValue();
 
-                buildAndSend(httpRequest, out);
-
-                Response<?> response = buildResponse(httpRequest, in);
-                HttpHeader locationHeader = response.getHeaders().get(HttpHeaderName.LOCATION);
-                String redirectLocation = (locationHeader == null) ? null : locationHeader.getValue();
-
-                if (redirectLocation != null) {
-                    if (redirectLocation.startsWith("http")) {
-                        httpRequest.setUrl(redirectLocation);
-                    } else {
-                        httpRequest.setUrl(new URL(httpRequest.getUrl(), redirectLocation));
-                    }
-                    return sendPatchRequest(httpRequest, sslSocketFactory);
+            if (redirectLocation != null) {
```

Comment:
Any reason we are handling redirect different in PATCH vs all other HTTP methods?

Reply:
Didn't look into the redirect scenario much since this was existing code.

Reply:
Ah, if that's the case let's leave it but file an issue as this is a bit odd as no other HttpClient implementations do this for PATCH

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/40030